### PR TITLE
UT_ASSERTs moved to test/c_api

### PR DESCRIPTION
**Pull request by EuphoricThinking**
### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

File: test/c_api/ut_asserts.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#ifndef UMF_UT_ASSERTS_H
```
[the remark this answers is not in the document]
Comment (EuphoricThinking) [reply test/c_api/ut_asserts.h]: Done

File: test/c_api/ut_asserts.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#define UT_ASSERTne(lhs, rhs)                                                  \
+    ((void)(((lhs) != (rhs)) ||                                                \
+            (UT_FATAL("%s:%d %s - assertion failure: %s (0x%llx) != %s "       \
+                      "(0x%llx)",                                              \
+                      __FILE__, __LINE__, __func__, #lhs,                      \
+                      (unsigned long long)(lhs), #rhs,                         \
+                      (unsigned long long)(rhs)),                              \
+             0)))
+
+#endif /* UMF_UT_ASSERTS_H */
```
[the remark this answers is not in the document]
Comment (EuphoricThinking) [reply test/c_api/ut_asserts.h]: Done

File: test/c_api/ut_asserts.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
+#ifndef UMF_UT_ASSERTS_H
+#define UMF_UT_ASSERTS_H 1
+
```
[the remark this answers is not in the document]
Comment (EuphoricThinking) [reply test/c_api/ut_asserts.h]: Done

File: test/c_api/disjoint_pool.c [hunk 4]
```diff
@@ -8,6 +8,8 @@
 #include "provider_null.h"
 #include "test_helpers.h"
```
[the remark this answers is not in the document]
Comment (EuphoricThinking) [reply test/c_api/disjoint_pool.c]: Removed

File: test/c_api/test_ut_asserts.h [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * Copyright (C) 2023-2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+/*
+ The project uses GTEST framework for testing, which is not supported in C. Therefore, there have been introduced customised asserts for C API test. These asserts should not be used in other purposes than for testing C files
```
Comment (lukaszstolarczuk) [inline test/c_api/test_ut_asserts.h:11]: heh, this is a little too descriptive - comments should rather be short and concise.

also, please break lines. Each line should fit "in one screen", meaning, usually around 100-120 chars per line (we don't have strict limit for comments, I guess).

example:
```
/*
 * This project uses GTEST framework for testing, which is not supported in C.
 * These custom asserts can and should ONLY be used in C API tests!
 */
```

Comment (EuphoricThinking) [reply test/c_api/test_ut_asserts.h]: Done:)

**Review by lplewa — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**